controller/src/main/java/io/pravega/controller/server/eventProcessor/ControllerEventProcessors.java

```diff
@@ -156,6 +158,8 @@ public ControllerEventProcessors(final String host,
                 new SealStreamTask(streamMetadataTasks, streamTransactionMetadataTasks, streamMetadataStore, executor),
                 new DeleteStreamTask(streamMetadataTasks, streamMetadataStore, bucketStore, executor),
                 new TruncateStreamTask(streamMetadataTasks, streamMetadataStore, executor),
+                new AddSubscriberTask(streamMetadataTasks, streamMetadataStore, executor),
```

Comment:
do you need to add a subscriber asynchronously? that can be done inline for the call as it should be a single metadata update -- a new subscriber id as a record in some table, right?

Reply:
A 'StreamSubscriberRecord' has the 'subscriberId' and 'SubscriberConfiguration' for multiple subscribers.
And addSubscriber and updateStreamCutForSubscriber, updates the same metadata record, hence it is necessary to handle this asynchronously.